### PR TITLE
fix(inspector): derive an addon's parent services from its functions

### PR DESCRIPTION
**Pull request by yasserf**
## The problem

Service aggregation is driven by `usedFunctions`, which is populated from wirings. An addon's functions are its published surface — a consumer calls them, nothing wires them locally — so `usedFunctions` is empty during an addon's own build and **no function services were aggregated at all**.

`requiredParentServices` was then derived solely from the second parameter of `pikkuAddonServices`: what the factory *reads*, not what its functions *use*.

The visible cost: an addon consuming `kysely` in all fifteen of its functions declared it nowhere. The generated map said `'kysely': false`, the consuming project was never told to supply it, the addon lost it at runtime, and every query failed on the first call. The only way out was listing the missing names under `forceRequiredServices` — a detection escape hatch (`usedServices.add`), never meant to carry a contract.

## The change

An addon build now:

1. **aggregates services from every declared function**, not just wired ones;
2. **treats what it uses but does not build as parent-provided.**

"Builds" is read from the object literals the factory returns, minus anything it took off the parent bag first — a forwarded service is not a created one, which is how `@pikku/addon-admin` returns `scopeService`, `credentialService`, `audit` and `auth` without claiming to own them. Names destructured from the second parameter *in the function body* are recognised alongside parameter destructuring.

Two new `InspectorState` fields carry this: `addonCreatedServices` and `addonServicesFactorySeen`.

## Effect

On an addon whose functions all use `kysely`:

```diff
-export const requiredParentServices = ["secrets","variables"] as const
+export const requiredParentServices = ["secrets","variables","kysely"] as const
-  'kysely': false,
+  'kysely': true,
```

and its `forceRequiredServices` entry becomes redundant — output is identical with and without it.

## Verification

- 745/745 inspector tests pass.
- Regenerating `@pikku/addon-admin`, `@pikku/addon-console` and `@pikku/addon-graph` leaves their tracked `.pikku` output **unchanged** — the forwarded-vs-created distinction is what keeps `addon-admin` stable.
- The addon that surfaced this now builds and passes its own 184 tests against the corrected contract.

Note: the local `@pikku/cli` build fails on 18 pre-existing errors in stale generated `.pikku/*.gen.ts` files, unrelated to any file in this PR.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved detection of services provided by add-ons and those required from their parent applications.
  * Add-on functions now correctly contribute to required service calculations.
  * Improved support for services accessed through destructuring and returned service objects.
  * Preserved compatibility when restoring previously saved inspection data.

### CODE REVIEW

File: packages/inspector/src/add/add-file-with-factory.ts [hunk 1]
```diff
@@ -13,6 +13,102 @@ const wrapperFunctionMap: Record<string, string> = {
...
+
+  const collectBinding = (pattern: ts.ObjectBindingPattern) => {
+    for (const elem of pattern.elements) {
+      const name =
+        elem.propertyName && ts.isIdentifier(elem.propertyName)
+          ? elem.propertyName.text
+          : ts.isIdentifier(elem.name)
+            ? elem.name.text
+            : undefined
+      if (name) forwarded.push(name)
```
[the remark this answers is not in the document]
Comment (yasserf) [reply packages/inspector/src/add/add-file-with-factory.ts:37]: Fixed in be966d598 — both shapes were real. Rest binding elements are now skipped via `elem.dotDotDotToken`, and parenthesised expressions are unwrapped wherever a returned expression is read, not only for a concise arrow body.

Verifier coverage added in `packages/inspector/src/add/add-file-with-factory.test.ts`, running through `inspect` end to end: a body-destructured `kysely` is forwarded while a factory-built service is not, a rest binding is excluded, a parenthesised return still yields its created services, and the three new state fields survive a serialize/deserialize round trip. The two edge-case tests fail on the previous commit and pass on this one.

File: packages/inspector/src/utils/post-process.ts [hunk 2]
```diff
@@ -381,6 +381,35 @@ export function aggregateRequiredServices(
...
+    for (const service of requiredServices) {
+      if (
+        !createdSet.has(service) &&
+        !internalServices.has(service) &&
+        !defaultServices.has(service)
+      ) {
+        parentServices.add(service)
+      }
+    }
+    state.addonRequiredParentServices = [...parentServices]
```
[the remark this answers is not in the document]
Comment (yasserf) [reply packages/inspector/src/utils/post-process.ts:411]: Fixed in be966d598 — both shapes were real. Rest binding elements are now skipped via `elem.dotDotDotToken`, and parenthesised expressions are unwrapped wherever a returned expression is read, not only for a concise arrow body.

Verifier coverage added in `packages/inspector/src/add/add-file-with-factory.test.ts`, running through `inspect` end to end: a body-destructured `kysely` is forwarded while a factory-built service is not, a rest binding is excluded, a parenthesised return still yields its created services, and the three new state fields survive a serialize/deserialize round trip. The two edge-case tests fail on the previous commit and pass on this one.
